core/templates/components/question-directives/questions-list/questions-list.component.ts

```diff
@@ -273,51 +273,69 @@ export class QuestionsListComponent implements OnInit, OnDestroy {
     this.windowRef.nativeWindow.location.hash = this.questionId;
   }
 
-  deleteQuestionFromSkill(
-      questionId: string, skillDescription: string): void {
+  async isQuestionDeletionAllowed(): Promise<boolean> {
+    let questionDeletionIsAllowed = true;
     if (!this.canEditQuestion) {
       this.alertsService.addWarning(
         'User does not have enough rights to delete the question');
-      return;
+      questionDeletionIsAllowed = false;
+      return questionDeletionIsAllowed;
     }
 
-    this.deletedQuestionIds.push(questionId);
-    // For the case when, it is in the skill editor.
-    if (this.allSkillSummaries.length === 0) {
-      this.editableQuestionBackendApiService.editQuestionSkillLinksAsync(
-        questionId, [
-          {
-            id: this.selectedSkillId,
-            task: 'remove'
-          } as SkillLinkageModificationsArray
-        ]
-      ).then(() => {
-        this.questionsListService.resetPageNumber();
-        this.questionsListService.getQuestionSummariesAsync(
-          this.selectedSkillId, true, true);
-        this.alertsService.addSuccessMessage('Deleted Question');
-        this._removeArrayElement(questionId);
-      });
-    } else {
-      this.allSkillSummaries.forEach((summary) => {
-        if (summary.getDescription() === skillDescription) {
-          this.editableQuestionBackendApiService.editQuestionSkillLinksAsync(
-            questionId, [
-              {
-                id: summary.getId(),
-                task: 'remove'
-              } as SkillLinkageModificationsArray
-            ]
-          ).then(() => {
-            this.questionsListService.resetPageNumber();
-            this.questionsListService.getQuestionSummariesAsync(
-              this.selectedSkillId, true, true);
-            this.alertsService.addSuccessMessage('Deleted Question');
-            this._removeArrayElement(questionId);
-          });
+    let numberOfQuestions = this.questionSummariesForOneSkill.length;
+    await this.skillBackendApiService
+      .getTopicNamesWithGivenSkillAssignedForDiagnosticTest(
+        this.selectedSkillId).then((topicNames) => {
+        if ((topicNames.length > 0) && (
+          numberOfQuestions <=
+            AppConstants.MINIMUM_QUESTION_COUNT_FOR_A_DIAGNOSTIC_TEST_SKILL)) {
+          let alertMessge = (
```

Comment:
alertMessage is misspelled.

Also, see my [previous comment](https://github.com/oppia/oppia/pull/15742#discussion_r939785405): I wrote "Shouldn't this be in a modal (with links to the corresponding topic editors)?" You wrote "updated" but this still seems to be an alert message?

Reply:
Alert is now removed, and a confirmation modal will be presented before removing a question from the skill.